Solves issue #1715 of zoom control

4shutosh opened the pull request:
Fixes #1715 

Added missing condition in `SliderPrefernce.java` 

![zoom_issue_solution1](https://user-images.githubusercontent.com/53831987/73295024-e5d29980-422c-11ea-8242-8b197881caa7.gif)

abdulwd commented on core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java:
```diff
@@ -168,12 +174,22 @@ protected void onBindView(View view) {
 
     TextView titleView = view.findViewById(android.R.id.title);
     TextView summaryTV = view.findViewById(android.R.id.summary);
     if (!enabled) {
```
As you have appended `else` the negative control flow can be avoided:
```java
if (enabled) {
  // some statements
} else {
  // other statements
}
```
Negative flow is difficult to read. :(

abdulwd commented on core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java:
```diff
@@ -168,12 +174,22 @@ protected void onBindView(View view) {
       titleView.setTextColor(Color.GRAY);
       summaryTV.setTextColor(Color.GRAY);
+    }else{
+      if(nightMode_enabled){
```
There should be spaces before `(` & after `)`.
Check out our code style guidelines in the README of this repo.

macgills replied:
you can press `ctrl+alt+l` to format, our codestyle is part of this project. I generally recommend setting up a macro to format,optimise imports and save bound to ctrl+s

4shutosh replied:
I'll fix the spaces and use proper codestyle wherever required.

macgills commented on core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java:
```diff
@@ -40,6 +43,9 @@
 
   private TextView mMessage;
 
+  @Inject
```
If you annotate a field `@Inject` then you have to call `TheCorrectComponent.inject(this)` for the object to be injected.

In this instance it would be `CoreApp.getCoreComponent().inject(this)` and that should be called in the constructor

macgills commented on core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java:
```diff
@@ -168,12 +174,22 @@ protected void onBindView(View view) {
+        summaryTV.setTextColor(Color.WHITE);
+      }else{
+        titleView.setTextColor(Color.BLACK);
+        summaryTV.setTextColor(Color.BLACK);
```
This whole logic can have the scope reduced, what actually happens here is you decide the textColor and then repeat the text setting code. 
Something like
```
int color = resolveColorFromState()
titleView.setTextColor(color);
summaryTV.setTextColor(color);

private int resolveColorFromState(){
  if(isEnabled){
    return nightModeConfig.isNightModeActive ? WHITE : BLACK
  }
  return GRAY;
}
```
and that is just a rough outline, AS will probably offer to reduce that even further

macgills commented:
@abdulwd re:Themes I would think if we just deleted the code that sets the text color in SliderPreference and just set the enabled state of the view then the text should behave correctly

abdulwd commented:
> @abdulwd re:Themes I would think if we just deleted the code that sets the text color in SliderPreference and just set the enabled state of the view then the text should behave correctly

@4shutosh can you please try this?

4shutosh commented:
> > @abdulwd re:Themes I would think if we just deleted the code that sets the text color in SliderPreference and just set the enabled state of the view then the text should behave correctly
> 
> @4shutosh can you please try this?

Yes I'll surely try this and get back to you.

4shutosh commented:
> @abdulwd re:Themes I would think if we just deleted the code that sets the text color in SliderPreference and just set the enabled state of the view then the text should behave correctly

> > @abdulwd re:Themes I would think if we just deleted the code that sets the text color in SliderPreference and just set the enabled state of the view then the text should behave correctly
> 
> @4shutosh can you please try this?

Hey!
This defintely worked.
Unused statements have been removed.

abdulwd commented on core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java:
```diff
@@ -171,9 +170,12 @@ private void setGrayState(View view) {
+      summaryTV.setEnabled(true);
+    } else {
+      titleView.setEnabled(false);
+      summaryTV.setEnabled(false);
```
Can you try removing these statements & check if it works?
Otherwise, it can be shortened to:
```java
titleView.setEnabled(enabled);
summaryTV.setEnabled(enabled);
```
;)

4shutosh replied:
> Can you try removing these statements & check if it works?
> Otherwise, it can be shortened to:
> 
> ```java
> titleView.setEnabled(enabled);
> summaryTV.setEnabled(enabled);
> ```
> 
> ;)

Yes i did try this as well.
It keeps the zoom level enabled irrespective of the Zoom controls toggle state.
If you insist I'll try again and get back to you! :)

4shutosh replied:
> Can you try removing these statements & check if it works?
> Otherwise, it can be shortened to:
> 
> ```java
> titleView.setEnabled(enabled);
> summaryTV.setEnabled(enabled);
> ```
> 
> ;)

Hey @abdulwd thee code you suggested is not working .
It keeps the zoom level enabled irrespective of the zoom controls toggle state.

macgills replied:
this cannot be logically true as the code
```
if(myBoolean){
setEnabled(true)
} else{
setEnabled(false)
}
```
is identical to
```
setEnabled(myBoolean)
```

Android studio should give you warning and even still I would say the compiler rewrites it

macgills commented on core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java:
```diff
@@ -171,9 +170,12 @@ private void setGrayState(View view) {
-    if (!enabled) {
-      titleView.setTextColor(Color.GRAY);
-      summaryTV.setTextColor(Color.GRAY);
+    if (enabled) {
```
just delete these, this whole method should be deleted. Setting the enabled state of the parent view should mean that state gets propagated downwards to its children and this already occurs in `CorePrefsFragment`
```
  private void setSliderState() {
    boolean enabled = getPreferenceManager().getSharedPreferences().getBoolean(
      PREF_ZOOM_ENABLED, false);
    if (enabled) {
      mSlider.setEnabled(true);
    } else {
      mSlider.setEnabled(false);
    }
  }
```
albeit in a very longwinded way could be 1 line
```
mSlider.setEnabled(sharedPrefUtil.getPrefZoomEnabled());
```

4shutosh replied:
> just delete these, this whole method should be deleted. Setting the enabled state of the parent view should mean that state gets propagated downwards to its children and this already occurs in `CorePrefsFragment`
> 
> ```
>   private void setSliderState() {
>     boolean enabled = getPreferenceManager().getSharedPreferences().getBoolean(
>       PREF_ZOOM_ENABLED, false);
>     if (enabled) {
>       mSlider.setEnabled(true);
>     } else {
>       mSlider.setEnabled(false);
>     }
>   }
> ```
> 
> albeit in a very longwinded way could be 1 line
> 
> ```
> mSlider.setEnabled(sharedPrefUtil.getPrefZoomEnabled());
> ```

@macgills This has been implemented in the latest commit.
Please review!

macgills replied:
All working great, thanks for the work

abdulwd commented:
@4shutosh Nice work!

4shutosh commented:
> @4shutosh Nice work!

I really look forward to contribute more to this repo!
